lib/recursive-methods.rb

```diff
@@ -1,49 +1,140 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+    if n < 0
+        raise ArgumentError, "Will only accept positive intergar"
+    end
+
+    return 1 if  n == 0 
+    
+    return n * factorial(n-1)
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def reverse(s)
```

Comment:
Since you are making a new string with each recursive call this is O(n^2) space complexity.